### PR TITLE
Issue #10924: Update GenericWhitespaceCheck to use code points

### PR DESCRIPTION
**Pull request by MUzairS15**
Issue #10924: Updated GenericWhitespaceCheck to use code points

Diff Regression config: https://gist.githubusercontent.com/MUzairS15/29772ece533f2a32313bd9baf085809d/raw/69aa032d1ed30f9996663b15a531729b6e4d35f9/config.xml

### CODE REVIEW

**Comment by MUzairS15** [comment]
Github, rebase

**Comment by MUzairS15** [comment]
Github, generate report

**Comment by MUzairS15** [comment]
```
src % cat config.xml Test.java 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <property name="severity" value="warning"/>

    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="TreeWalker">
        <module name="GenericWhitespace">
            
        </module>
    </module>
</module>

public class Test<V> {
    public <V> void methodName(V value) {

        /* 👉🏻  😆 */ Supplier<?> t = Test<V>::new; // no violation, but reported
        /* haha */ Supplier<?> t = Test<V>::new; // no violation

        Collections. /*da sd😆sd*/ <Object> emptySet(); // 2 violations but not reported
        Collections. /*da sdsd*/ <Object> emptySet(); // violations and reported

    }
}                                                                                                                                 
src % java -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[WARN] /Users/shabana/Documents/OpenSource/CheckStyleTest/src/Test.java:4:31: '>' is followed by an illegal character. [GenericWhitespace]
[WARN] /Users/shabana/Documents/OpenSource/CheckStyleTest/src/Test.java:8:34: '<' is preceded with whitespace. [GenericWhitespace]
[WARN] /Users/shabana/Documents/OpenSource/CheckStyleTest/src/Test.java:8:41: '>' is followed by whitespace. [GenericWhitespace]
Audit done.
```
@romani @nick-mancuso

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java [hunk 1]
```diff
@@ -228,7 +231,8 @@ private void processNestedGenerics(DetailAST ast, String line, int after) {
         //                                          ^
         //   should be whitespace if followed by & -+
         //
-        final int indexOfAmp = line.indexOf('&', after);
+        final int indexOfAmp = IntStream.range(after, line.length)
```
Comment (nick-mancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java:234]: Please wrap on each successive method call after this line.

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
One minor item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java (hunk 1 above)
Comment (MUzairS15) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java:234]: done

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java:234]: `.orElse(-1);` next line, as well.

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java:234]: Not done, please wrap`orElse`

Comment (MUzairS15) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java:234]: Better if I do tomorrow, not in state of mind😴

Comment (MUzairS15) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java:234]: done

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java:234]: Resolved.

**Review by nick-mancuso — APPROVED** [review]

**Review by romani — APPROVED** [review]

**Pull request merged**